modules/common/job/job.go

```diff
@@ -170,14 +170,22 @@ func (j *Job) DoJob(
 	}
 
 	// allow updating TTLSecondsAfterFinished even after the job is finished
-	_, err = controllerutil.CreateOrPatch(ctx, h.GetClient(), job, func() error {
-		job.Spec.TTLSecondsAfterFinished = j.job.Spec.TTLSecondsAfterFinished
-		return nil
-	})
-	if err != nil {
+	job, err = GetJobWithName(ctx, h, j.job.Name, j.job.Namespace)
+	if err != nil && !k8s_errors.IsNotFound(err) {
 		return ctrl.Result{}, err
 	}
 
+	if err == nil {
+		_, err = controllerutil.CreateOrPatch(ctx, h.GetClient(), job, func() error {
+			job.Spec.TTLSecondsAfterFinished = j.job.Spec.TTLSecondsAfterFinished
+			return nil
+		})
+
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
 	return ctrl.Result{}, nil
```

Comment:
may be this return path can be merged with the above condition, so that at the line 174, it will just return on any err != nill?

Reply:
Yes, we can exit early a L174 if there is any error. But we still need L189 for the case where the Job exists, and the TTL update happened successfully at L182. So we cannot reduce the number of returns.